sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -244,8 +244,8 @@ internal DataLakeFileSystemClient(Uri fileSystemUri, HttpPipelinePolicy authenti
         {
             options ??= new DataLakeClientOptions();
             _uri = fileSystemUri;
-            _blobUri = GetBlobUri(fileSystemUri);
-            _dfsUri = GetDfsUri(fileSystemUri);
+            _blobUri = new DataLakeUriBuilder(fileSystemUri).ToBlobUri();
+            _dfsUri = new DataLakeUriBuilder(fileSystemUri).ToDfsUri();
```

Comment:
Can we use the same UriBuilder for both?

Also applies to other places.

Reply:
No, because we are currently resetting the value of the Host property at the end of DataLakeUriBuilder.ToBlobUri() and .ToDfsUri()

Reply:
But wouldn't we flip it back when calling ToDfsUri?

Reply:
Yes, but this behavior seems weird to me.  Maybe we could make .ToBlobUri() and .ToDfsUri() static?

Reply:
We use instance methods for ToUri on the other builders, so I think it makes sense to be consistent.

Reply:
fixed.